File: app/src/main/java/it/unical/mat/embasp/base/Output.java
package it.unical.mat.embasp.base;

/** represents a generic output for a solver */
public class Output implements Cloneable {
	/** Variable in which results are stored */
	protected String output;
	/** The errors thrown by the solver */
	protected String errors;

	public Output() {
		output = new String();
	}

	public Output(final String initial_output) {
		output = initial_output;
	}

	public Output(final String out, final String err) {
		output = out;
		errors = err;
	}

	@Override
	public Object clone() throws CloneNotSupportedException {
		return super.clone();
	}

	public String getErrors() {
		return errors;
	}

	public String getOutput() {
		return output;
	}

	protected void parse() {
	}

	public void setErrors(final String err) {
		errors = err;
	}

	public void setOutput(final String output) {
		this.output = output;
	}
}

File: Python version/base/output.py
class Output(object):
    """Represents a generic output for a solver."""

    def __init__(self, output=None, errors=None):
        self._output = output  # Variable in which results are stored
        self._errors = errors  # The errors thrown by the solver

    def get_errors(self):
        """Get error string."""
        return self._errors

    def get_output(self):
        """Get output string."""
        return self._output

    def set_errors(self, errors):
        """Set error string."""
        self._errors = errors

    def set_output(self, output):
        """Set output string."""
        self._output = output

    def _parse(self):
        """This method have to be implemented by subclasses to parse a solver
        output."""
        pass
